File: 4주차/ain/programmers4_1.java
public class programmers4_1 {

    static class Solution {
        public static int solution(int n) {
            int answer = 0;
            for(int i = 1; i <= n; i++) {
                int sum = 0;
                int count = i;
                while(true) {
                    sum += count;
                    if(sum == n) {
                        answer++;
                        break;
                    }else if(sum > n) {
                        break;
                    }else {
                        count++;
                    }
                }
            }
            return answer;
        }
    }

    public static void main(String[] args) {
        int n = 15;
        System.out.println(Solution.solution(n));
    }
}

File: 4주차/yeonjin/programmers4_1.py
def solution(n):
    answer = 0
    if n == 1: 
        answer = 1

    for k in range(1, n):
        num = n + (k*k - k) / 2 
        a = (n/k) + (1-k)/2
        if a > 0 :
            if num > 0 and num % k == 0:
                answer += 1
    return answer

